### ISSUE TITLE
BGP AS Number in VRF Tier-0 Gateway

### ISSUE THREAD

**Issue by davidvanr** [vmware/terraform-provider-nsxt#1073]
### Is your feature request related to a problem? Please describe.

It is not possible to add the BGP AS Number of a VRF Tier-0 gateway through the nsxt_policy_tier0_gateway resource. 
Adding it with the resource nsxt_policy_bgp_config, is not working, the BGP_AS number in not taken into account for a Tier-0 VRF Gateway. 

### Describe the solution you'd like

Add a local_as_num param for the VRF configuration in nsxt_policy_tier0_gateway.

### Describe alternatives you've considered

Tried with below configuring but it's not working.

resource "nsxt_policy_bgp_config" "gw1" {
  gateway_path = nsxt_policy_tier0_gateway.vrf_gateway.path

  enabled               = true
  inter_sr_ibgp         = false
  local_as_num          = "65555"
  graceful_restart_mode = "HELPER_ONLY"

}

**Comment by salv-orlando** [comment]
Many thanks for reporting this issue!

It appears the provider was considering `local_as_num` as not supported for VRF Tier-0s. I checked that it is supported at least since 3.2.0. I pushed PR https://github.com/vmware/terraform-provider-nsxt/pull/1074 to address this, and in parallel checking whether any older NSX release did not support setting this attribute on VRF Tier-0s.

**Comment by liftconfig** [comment]
> Many thanks for reporting this issue!
> 
> It appears the provider was considering `local_as_num` as not supported for VRF Tier-0s. I checked that it is supported at least since 3.2.0. I pushed PR #1074 to address this, and in parallel checking whether any older NSX release did not support setting this attribute on VRF Tier-0s.

BGP ASN per T0 VRF was introduced in [NSX release 4.1.0](https://docs.vmware.com/en/VMware-NSX/4.1.0/rn/vmware-nsx-410-release-notes/index.html)

> BGP Autonomous System (AS) Number per Tier-0 VRF Gateway and BGP neighbor - This release introduces the support for configuring a different BGP ASN (Autonomous System Number) per Tier-0 VRF Gateway and also per BGP neighbor. Defining a separate ASN per VRF and BGP peer is an important feature for Service Provider and multi tenant topologies where end-customers will bring their own BGP ASN to the networking topologies.

**Comment by mr-arzt** [comment]
We would love to have this supported too.

**Comment by salv-orlando** [comment]
https://github.com/vmware/terraform-provider-nsxt/pull/1074 addressed this issue

**Comment by carmine73** [comment]
On documentation I read

  * `inter_sr_ibgp` - (Optional) A boolean flag to enable/disable inter SR IBGP configuration. Default is `true`. **This setting is not applicable to VRF-Lite Gateway**.

This option is available since on UI since 4.2.1, are you going to add also this setting?

**Comment by annakhm** [comment]
Thanks @carmine73, yes we can support this attribute for VRF as well.
```

**Comment by carmine73** [comment]
Thanks @annakhm, do you want me to open a new issue for that?
I see you've just created tag 3.8.2, I'm too late :-(

**Comment by annakhm** [comment]
Hi @carmine73, yes please open a new issue, it will be easier to track. Thanks!

**Comment by carmine73** [comment]
@annakhm new issue https://github.com/vmware/terraform-provider-nsxt/issues/1659 as requested
thanks
